make/firmware-common.mk

```diff
@@ -1,3 +1,28 @@
+# Check for RTOS
+ifeq ($(RTOS), CHIBIOS)
+  CDEFS += -DPIOS_INCLUDE_CHIBIOS
+  include $(APPLIBDIR)/ChibiOS/library.mk
+  SRC += board.c
+else ifeq ($(RTOS), FREERTOS)
+  CDEFS += -DPIOS_INCLUDE_FREERTOS
+  include $(PIOSCOMMONLIB)/FreeRTOS/library.mk
+else ifeq ($(RTOS), UNDEFINED_FOR_UAVOBJECT_LIB)
+  # Do nothing here, it's only included as an escape for the
+else
+  $(error [RTOS]: $(RTOS) is not a valid RTOS. Choose between either CHIBIOS or FREERTOS)
+endif
+
+## PIOS Hardware (STM32F1/3/4xx or POSIX)
+# `ifneq ($(or $(VAR1),$(VAR2)),)` is the Makefile code for `if defined(VAR1) || defined(VAR2)`
+ifneq ($(or $(PIOSSTM32FXXX),$(PIOSPOSIX)),)
+  ifeq ($(RTOS), CHIBIOS)
+    include $(PIOS)/$(TARGET_ARCHITECTURE)/library_chibios.mk
```

Comment:
I originally did this, but the `SIM_POSIX` caused some problems there because it's non-STM32, so I refactored into this approach. This also generally opens the door to non-STM32 chips.